runtime/ops/signal.rs

```diff
@@ -32,7 +40,21 @@ pub fn init(rt: &mut deno_core::JsRuntime) {
 #[cfg(unix)]
 /// The resource for signal stream.
 /// The second element is the waker of polling future.
-pub struct SignalStreamResource(pub Signal, pub Option<Waker>);
+struct SignalStreamResource {
+  signal: AsyncRefCell<Signal>,
+  cancel: CancelHandle,
+}
+
+#[cfg(unix)]
+impl Resource for SignalStreamResource {
+  fn name(&self) -> Cow<str> {
+    "signal".into()
+  }
+
+  fn close(self: Rc<Self>) {
+    self.cancel.cancel();
+  }
+}
```

Comment:
Aside: I think we should factor this all out into core in future work. I don't see the point in doing this for every resource implementation.

Reply:
This is tricky... some resources don't have `cancel` field; also if that was part of `deno_core` then we'd need to create a wrapper there. Let's postpone this for now